lib/m2ts/caption-stream.js

```diff
@@ -610,8 +638,23 @@ Cea708Stream.prototype.getPts = function(byteIndex) {
  */
 Cea708Stream.prototype.initService = function(serviceNum, i) {
   var self = this;
+  var serviceNames = Object.keys(this.serviceEncodings);
+  var serviceName;
+  var providedEncoding;
+  var encoding;
+
+  // Check if an encoding value was specified for this service
+  for (var i = 0; i < serviceNames.length; i++) {
+    serviceName = serviceNames[i];
+    providedEncoding = this.serviceEncodings[serviceName];
+
+    if (serviceName === 'SERVICE' + serviceNum) {
+      encoding = providedEncoding;
+      break;
+    }
+  }
```

Comment:
since `serviceEncodings` is a mapping of service names to encodings, you can use the `in` operator, to know whether the object has that property or not. So, we should be able to simplify this to be the following
```suggestion
  var serviceName = 'SERVICE' + serviceNum;
  if (serviceName in this.serviceEncodings) {
    encoding = this.serviceEncodings[serviceName];
  }
```

Reply:
Much better